python/ray/experimental/sgd/examples/cifar-tf-example.py

```diff
@@ -0,0 +1,217 @@
+
+    # Compute quantities required for feature-wise normalization
+    # (std, mean, and principal components if ZCA whitening is applied).
+    datagen.fit(x_train)
+    return datagen.flow(x_train, y_train, batch_size=batch_size)
+
+
+def data_augmentation_creator(batch_size):
+    (x_train, y_train), (x_test, y_test) = fetch_keras_data()
+    trainset = tf.data.Dataset.from_generator(
+        lambda: _make_generator(x_train, y_train, batch_size),
+        output_types=(tf.float32, tf.float32),
+        # https://github.com/tensorflow/tensorflow/issues/24520
+        output_shapes=(tf.TensorShape((None, None, None, None)),
+                       tf.TensorShape((None, 10))))
+    trainset = trainset.repeat()
+
+    test_dataset = tf.data.Dataset.from_tensor_slices((x_test, y_test))
+    test_dataset = test_dataset.repeat().batch(batch_size)
+    return trainset, test_dataset
+
+
+def train_example(num_replicas=1, batch_size=512, epochs=1, use_gpu=False):
+    trainer = TFTrainer(
+        model_creator=create_model,
+        data_creator=data_augmentation_creator,
+        num_replicas=num_replicas,
+        use_gpu=use_gpu,
+        config={
+            "verbose": True,
+            "fit_config": {
+                "steps_per_epoch": 60000 // batch_size
+            },
+            "evaluate_config": {
+                "steps": 10000 // batch_size,
+            }
+        },
+        batch_size=batch_size)
+
+    for i in range(epochs):
+        train_stats1 = trainer.train()
+        train_stats1.update(trainer.validate())
+        print("iter {}:".format(i), train_stats1)
+
+    model = trainer.get_model()
+    trainer.shutdown()
+    return model
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--redis-address",
+        required=False,
+        type=str,
+        help="the address to use for Redis")
+    parser.add_argument(
+        "--num-replicas",
+        "-n",
+        type=int,
+        default=1,
+        help="Sets number of replicas for training.")
+    parser.add_argument(
+        "--batch-size",
+        type=int,
+        default=512,
+        help="Sets number of replicas for training.")
+    parser.add_argument(
+        "--num-epochs",
+        type=int,
+        default=1,
+        help="Sets number of replicas for training.")
+    parser.add_argument(
+        "--use-gpu",
+        action="store_true",
+        default=False,
+        help="Enables GPU training")
+
+    args, _ = parser.parse_known_args()
+    ray.init(redis_address=args.redis_address)
+
+    batch_size = args.batch_size
+
+    model = train_example(
+        args.num_replicas,
+        batch_size=batch_size,
+        epochs=args.num_epochs,
+        use_gpu=args.use_gpu)
+    dataset, test_dataset = data_augmentation_creator(batch_size=batch_size)
+    model.fit(dataset, steps_per_epoch=60000 // batch_size, epochs=1)
+    scores = model.evaluate(test_dataset, steps=10000 // batch_size)
```

Comment:
TODO: test both data_augmentation and normal versions. Set smoke test, which only takes 10 steps.